File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.java
import java.util.Arrays;

class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        
        int[] num = new int[number];
        for (int i =0; i<number; i++){
            int cnt = 0; 
            for (int j = 1; j*j <= (i+1); j++){
                if(j*j == i+1){
                    cnt++;
                }
                else if ((i+1) % j ==0){
                    cnt+=2;
                }
            }
            num[i] = cnt;
        }
        for (int n : num){
            if (n > limit){
                answer += power;
            }
            else{
                answer += n;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    for i in range(number):
            cnt = 0
            for j in range(1, int((i + 1)**0.5) + 1):
                if j * j == i + 1:
                    cnt += 1
                elif (i + 1) % j == 0:
                    cnt += 2
            if cnt > limit:
                answer += power
            else:
                answer += cnt
    return answer